cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala

```diff
@@ -174,8 +174,20 @@ private[akka] final class BootstrapCoordinator(discovery: ServiceDiscovery,
       discoverContactPoints()
 
     case ServiceDiscovery.Resolved(_, contactPoints) ⇒
-      log.info("Located service members based on: [{}]: [{}]", lookup, contactPoints.mkString(", "))
-      onContactPointsResolved(contactPoints)
+      val filteredContactPoints: Iterable[ResolvedTarget] =
+        if (lookup.portName.isDefined)
+          contactPoints
+        else
+          contactPoints.groupBy(_.host).flatMap {
+            case (host, immutable.Seq(singleResult)) =>
+              immutable.Seq(singleResult)
```

Comment:
this may not be the akka management port, could be a user http port if the pod label selector picks up pods from different deployments. Not the normal but user could easily do it